cirq-core/cirq/ops/gateset.py

```diff
@@ -0,0 +1,309 @@
+# Copyright 2021 The Cirq Developers
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Functionality for grouping and validating Cirq Gates"""
+
+from typing import Any, Callable, Dict, FrozenSet, List, Optional, Type, TYPE_CHECKING, Union
+from cirq.ops import global_phase_op, op_tree, raw_types
+from cirq import protocols, value
+
+if TYPE_CHECKING:
+    import cirq
+
+
+@value.value_equality(distinct_child_types=True)
+class GateFamily:
+    """Wrapper around gate instances/types describing a set of accepted gates.
+
+    GateFamily supports initialization via
+        a) Non-parameterized instances of `cirq.Gate` (Instance Family).
+        b) Python types inheriting from `cirq.Gate` (Type Family).
+
+    By default, the containment checks depend on the initialization type:
+        a) Instance Family: Containment check is done by object equality.
+        b) Type Family: Containment check is done by type comparison.
+
+    For example:
+        a) Instance Family:
+            >>> gate_family = cirq.GateFamily(cirq.X)
+            >>> assert (cirq.X in gate_family) == True
+            >>> assert (cirq.X ** sympy.Symbol("theta") not in gate_family) == False
```

Comment:
Should this be `== True` ?

Reply:
Yeah, thanks for catching this. I think I'll just remove the `==`'s and keep asserts with `in` and `not in` constructs.